kore/src/Kore/Builtin/Bool.hs

```diff
@@ -163,48 +165,63 @@ builtinFunctions =
     xor a b = (a && not b) || (not a && b)
     implies a b = not a || b
 
+data UnifyBool = UnifyBool
+    { bool1, bool2 :: InternalBool
+    }
+
+matchBools ::
+    TermLike RewritingVariableName ->
+    TermLike RewritingVariableName ->
+    Maybe UnifyBool
+matchBools first second
+    | InternalBool_ bool1 <- first
+      , InternalBool_ bool2 <- second =
+        Just $ UnifyBool bool1 bool2
```

Comment:
If we are going to use name shadowing, let's do it properly with named field puns:

```suggestion
        Just $ UnifyBool{bool1, bool2}
```